tests/std/tests/P0896R4_ranges_subrange/test.compile.pass.cpp

```diff
@@ -807,7 +807,12 @@ namespace test_subrange {
     inline constexpr bool is_subrange<subrange<I, S, K>> = true;
 
     template <class T>
-    inline constexpr auto kind_of = illformed<T>();
+    struct ill_formed {
+        static_assert(always_false<T>);
+    };
+
+    template <class T>
+    inline constexpr auto kind_of = ill_formed<T>{};

```

Comment:
No change requested: in isolation, this is an occurrence of the "auto = temporary" pattern that I consistently object to as being overly convoluted and verbose. However, in context, this is fine - it is symmetrical with the well-formed specialization below, and here we don't actually care that the type is `ill_formed<T>` because it's going to be an error anyways.

Reply:
I agree that symmetry with the other partial specialization of `kind_of` trumps AAA here - the result is clearer overall.